File: go/activities.go
package starter

import (
	"errors"
	"fmt"
)

type Activities struct{}

func (a *Activities) StepOne(input string) (string, error) {
	fmt.Println("--> [Step One]  Input: " + input)
	return input, nil
}

func (a *Activities) StepTwo(input string) (string, error) {
	hasBug := true

	if hasBug {
		return "oops", errors.New("oops error. Hard to fix. Be warned.")
	} else {
		fmt.Println("--> [Step Two]  Input: " + input)
	}

	return input, nil
}

func (a *Activities) StepThree(input string) (string, error) {
	fmt.Println("--> [Step Three]  Input: " + input)
	return input, nil
}

File: typescript/src/activities.ts
export async function stepOne(step: string): Promise<string> {
  console.log(`Running step ${step}!`)
  return "One done!";
}

export async function stepTwo(step: string): Promise<string> {
  console.log(`Running step ${step}!`)

  // TODO FIXME
  const hasBug = true

  if (hasBug) {
    throw new Error("‼️ ERROR!!!! Really hard to fix. 😜")
  }
  return "Two done!";
}

export async function stepThree(step: string): Promise<string> {
  console.log(`Running step ${step}!`)
  return "Three done!";
}
